Assets/Scripts/SS3D/Systems/Entities/Humanoid/Ragdoll.cs

```diff
@@ -0,0 +1,334 @@
+            }
+        }
+
+        private void WalkingBehavior() { }
+        /// <summary>
+        /// Cast knockdown, that isn't going to expire until Recover()
+        /// </summary>
+        [ServerRpc(RequireOwnership = false)]
+        public void KnockdownTimeless()
+        {
+            _isKnockdownTimed = false;
+            IsKnockedDown = true;
+        }
+        /// <summary>
+        /// Knockdown the character for some time.
+        /// </summary>
+        /// <param name="seconds"></param>
+        [ServerRpc(RequireOwnership = false)]
+        public void Knockdown(float seconds)
+        {
+            _isKnockdownTimed = true;
+            _knockdownTimer += seconds;
+            IsKnockedDown = true;
+        }
+        private void Knockdown()
+        {
+            _currentState = RagdollState.Ragdoll;
+            ToggleController(false);
+            ToggleAnimator(false);
+            
+            if (!IsOwner) return;
+            Vector3 movement = _humanoidLivingController.TargetMovement * 3;
+            ToggleKinematic(false);
+            foreach (Transform part in _ragdollParts)
+            {
+                part.GetComponent<Rigidbody>().AddForce(movement, ForceMode.VelocityChange);
+            }
+        }
+
+        private void RagdollBehavior()
+        {
+            // Only the owner handles ragdoll's physics
+            if (!IsOwner) return;
+            AlignToHips();
+        }
+        /// <summary>
+        /// Adjust player's position and rotation
+        /// </summary>
+        private void AlignToHips()
+        {
+            IsFacingDown = _hips.transform.forward.y < 0;
+            Vector3 originalHipsPosition = _hips.position;
+            Vector3 newPosition = _hips.position;
+            // Get the lowest position
+            if (Physics.Raycast(transform.position, Vector3.down, out RaycastHit hitInfo))
+            {
+                // It's important to use + 0.0051f, because otherwise the player gets stuck
+                newPosition.y = hitInfo.point.y + 0.0051f;
+            }
+            _character.position = newPosition;
+            _hips.position = originalHipsPosition;
+            
+            Vector3 desiredDirection = _hips.up * (IsFacingDown ? 1 : -1);
+            desiredDirection.y = 0;
+            desiredDirection.Normalize();
+            Quaternion originalHipsRotation = _hips.rotation;
+            Vector3 rotationDifference = Quaternion.FromToRotation(transform.forward, desiredDirection).eulerAngles;
+            // Make sure that rotation is only around Y axis
+            rotationDifference.x = 0;
+            rotationDifference.z = 0;
+            transform.rotation *= Quaternion.Euler(rotationDifference);
+            _hips.rotation = originalHipsRotation;
+        }
+        /// <summary>
+        /// Start of moving bones to their first positions in StandUp animation
+        /// </summary>
+        private void BonesReset()
```

Comment:
doc not clear enough, is it doing some kind of interpolation ? it moves from the ragdoll state to the bones position in the first frame of the standing up animation ?